v1/topdown/strings.go

```diff
@@ -310,14 +310,14 @@ func builtinSubstring(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Ter
 
 	sbase := string(base)
 	if sbase == "" {
-		return iter(ast.InternedEmptyString)
+		return iter(ast.InternedStringTerm.Get(""))
```

Comment:
Maybe we could keep `InternedEmptyString` and spare us the lookup. `InternedEmptyString` could, in turn, also be added to `InternedStringTerm`.